tiled/adapters/array.py

```diff
@@ -41,15 +53,16 @@ def __init__(self, array, *, chunks=None, metadata=None, dims=None, specs=None):
         self._metadata = metadata or {}
         self._dims = dims
         self.specs = specs or []
+        self.references = references or []
 
     @classmethod
-    def from_array(cls, array, *, chunks=None, metadata=None, dims=None, specs=None):
+    def from_array(cls, data, *, metadata=None, dims=None, specs=None, references=None):
         return cls(
-            numpy.asarray(array),
-            chunks=chunks,
+            dask.array.from_array(data),
```

Comment:
Ditto here. Nothing in this section should change: only adding `references` at the end.